components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/util/WebSubHubAdapterUtil.java

```diff
@@ -0,0 +1,429 @@
+                    }
+                } else {
+                    String errorResponseBody;
+                    if (LoggerUtils.isDiagnosticLogsEnabled()) {
+                        DiagnosticLog.DiagnosticLogBuilder diagnosticLogBuilder = new DiagnosticLog.DiagnosticLogBuilder(
+                                WebSubHubAdapterConstants.LogConstants.WEB_SUB_HUB_ADAPTER,
+                                WebSubHubAdapterConstants.LogConstants.ActionIDs.PUBLISH_EVENT);
+                        diagnosticLogBuilder
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.URL, url)
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.TENANT_DOMAIN,
+                                        eventContext.getTenantDomain())
+                                .inputParam(WebSubHubAdapterConstants.LogConstants.InputKeys.TOPIC, topic)
+                                .resultMessage("Failed to publish event data to WebSubHub.")
+                                .resultStatus(DiagnosticLog.ResultStatus.FAILED)
+                                .logDetailLevel(DiagnosticLog.LogDetailLevel.INTERNAL_SYSTEM);
+                        LoggerUtils.triggerDiagnosticLogEvent(diagnosticLogBuilder);
+                    }
+                    try {
+                        errorResponseBody = EntityUtils.toString(response.getEntity());
+                        log.error("WebHubSub event publisher received " + responseCode + " code. Response data: " + errorResponseBody);
+                    } catch (IOException e) {
+                        log.error("Error while reading WebSubHub event publisher response. ", e);
+                    }
+                }
+            }
+
+            @Override
+            public void failed(final Exception ex) {
+
+                handleResponseCorrelationLog(request, requestStartTime, WebSubHubCorrelationLogUtils.RequestStatus.FAILED.getStatus(),
+                        ex.getMessage());
+                log.error("Publishing event data to WebSubHub failed. ", ex);
+            }
+
+            @Override
+            public void cancelled() {
+
+                handleResponseCorrelationLog(request, requestStartTime, WebSubHubCorrelationLogUtils.RequestStatus.CANCELLED.getStatus());
+                log.error("Publishing event data to WebSubHub cancelled.");
+            }
+        });
+
+    }
+
+    /**
+     * Invoke the WebSub Hub to register, unregister the topics.
+     *
+     * @param topic                     topic name.
+     * @param webSubHubBaseUrl          WebSub Hub base URL.
+     * @param operation                 whether to register, deregister the topic.
+     * @throws IOException              on errors while communicating with WebSub hub.
+     * @throws WebSubAdapterException   on client or server related errors.
+     */
+    public static void makeTopicMgtAPICall(String topic, String webSubHubBaseUrl, String operation)
```

Comment:
This also can be simplified. Moreover, client execution can be taken to a separate class IMO without repeating

Reply:
We can simplify this, but reusing the same code for publishing API call may need to check. The reason for 2 separate implementation is for handling separate exception messages and logs specific to each implementation.

Reply:
Adressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/8fa9c3addd4f16e4b35ffb0a8e7c96a28c089382